### ISSUE TITLE
Keep one-line ctors

### ISSUE THREAD

**Issue by mwoehlke-kitware** [uncrustify/uncrustify#1983]
This is both a matter of "do we have an option?" (I think we don't) and "change `forUncrustifySources.cfg`.

#1852 wants to introduce some code like:

```c++
   inline flags() : m_i{0} {}
   inline flags(Enum flag) : m_i{static_cast<int>(flag)} {}
   inline flags(const flags &other) : m_i{other.m_i} {}

   inline flags &operator=(const flags &other)
   { m_i = other.i; return(*this); }

   inline flags &operator&=(int mask)
   { m_i &= mask; return(*this); }
   inline flags &operator&=(unsigned int mask)
   { m_i &= mask; return(*this); }
```
...but uncrustify (with `forUncrustifySources.cfg`) wants to add a newline before the `:` of the ctors, which... does not seem beneficial to me. We have `nl_class_leave_one_liners`, which seems to be responsible for keeping the ctor bodies unsplit (good!), but I don't see an equivalent for suppressing `nl_constr_colon`.

What I'd like to add (regardless of what uncrustify wants for its own sources, because I want it for my own code) is an option to suppress `nl_constr_colon` if the ctor prototype and initializers are one line, but I'll also take the opportunity to ask, would that be acceptable for use in uncrustify's own sources? (If not, is there any alternate implementation that *would* be acceptable?)

**Comment by CDanU** [comment]
Having a new option for that would obviously be nice.
I don't really mind the real oneliners (first three) as long as the initializer list is short. I am however against having the `{` and `}` being on the same line if it begins with the `{`, especially if there is more than one command inside the braces.

**Comment by mwoehlke-kitware** [comment]
> For Uncrustify I don't really mind the real oneliners (first three) as long as the initializer list is short.

Right... I think I'm on the same page with "short", which is why I suggested "if the ctor prototype and initializers are one line" (which seems like a plausibly enforceable definition of "short").

> I am however against having the `{` and `}` being on the same line if it begins with the `{`, especially if there is more than one command inside the braces.

Well, that's sort of off topic :slightly_smiling_face:, since `forUncrustifySources.cfg` currently allows that. Those lines are shown here only for context; this issue is really only about the first three lines. That said, I'm not totally unsympathetic to that view, although I think here it would make the file unnecessarily lengthy. (To wit: *the entire class* consists of methods like that. I'm not convinced that ballooning the class definition by almost 3x is worth the alleged readability improvement in this case. But again, OT...)

**Comment by mwoehlke-kitware** [comment]
Actually, realizing now that I can simplify this a bunch with `= default` (the provenance of this code predates C++11), I think I can live with the newline in the one remaining ctor for uncrustify itself (well, I can with #1984 :slightly_smiling_face:).

I still believe I'm going to want this, eventually, for my own code, but #1852 can live without it...

**Comment by mihaipopescu** [comment]
> I am however against having the { and } being on the same line if it begins with the {, **especially if there is more than one command inside the braces**.

Precisely. We have done a lot of work to preserve one-liners for empty body functions like this and prefer the more compact style. There's no reason why you'd want the _ballooning_ anywhere. :)

As I said in the related PR (forgot the ref number), IMO it should be better to have another option that makes this rule less strict.
